commit bdf86d0e6ca307de8c85e9363b31ca8c86c0b0c7
Author: Russ Weight <russell.h.weight@intel.com>
Date:   Mon Jun 6 09:00:35 2022 -0700

    fpga: m10bmc-sec: create max10 bmc secure update
    
    Create a sub-driver for the FPGA Card BMC in order to support secure
    updates.  This patch creates the Max10 BMC Secure Update driver and
    provides sysfs files for displaying the root entry hashes (REH) for the
    FPGA static region (SR), the FPGA Partial Reconfiguration (PR) region,
    and the card BMC.
    
    The Intel MAX10 BMC Root of Trust (RoT) requires that all BMC Nios firmware
    and FPGA images are authenticated using ECDSA before loading and executing
    on the card. Code Signing Keys (CSK) are used to sign images. CSKs are
    signed by a root key. The root entry hash is created from the root public
    key.
    
    The RoT provides authentication by storing an REH bitstream to a write-once
    location. Image signatures are verified against the hash.
    
    Reviewed-by: Tom Rix <trix@redhat.com>
    Tested-by: Tianfei Zhang <tianfei.zhang@intel.com>
    Signed-off-by: Russ Weight <russell.h.weight@intel.com>
    Link: https://lore.kernel.org/r/20220606160038.846236-3-russell.h.weight@intel.com
    Signed-off-by: Xu Yilun <yilun.xu@intel.com>

diff --git a/MAINTAINERS b/MAINTAINERS
index a6d3bd9d2a8d..8f270acf0844 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -7807,6 +7807,13 @@ F:	Documentation/fpga/
 F:	drivers/fpga/
 F:	include/linux/fpga/
 
+INTEL MAX10 BMC SECURE UPDATES
+M:	Russ Weight <russell.h.weight@intel.com>
+L:	linux-fpga@vger.kernel.org
+S:	Maintained
+F:	Documentation/ABI/testing/sysfs-driver-intel-m10-bmc-sec-update
+F:	drivers/fpga/intel-m10-bmc-sec-update.c
+
 FPU EMULATOR
 M:	Bill Metzenthen <billm@melbpc.org.au>
 S:	Maintained